### DISCUSSION TITLE
Easun 2.2Kw KVA

### DISCUSSION THREAD

**Discussion by paolopa65** [softwarecrash/Solar2MQTT-ESP8266#88] [General]
Hi, has anyone tested the Easun 2.2Kw KVA series hybrid inverter?
Or on an Easun KVA series inverter

**Comment by all-solutions** [comment]
Hi,

as far as I know there is no KVA series.
KVA is more likely to be a unit of measurement (kilo-volt-ampere).
What exactly is the model label on the inverter?

**Reply by paolopa65** [reply]
Hi, you're right, I confused the lines, the model is: Isolar-SMH-II-2.2Kw by Easun

**Reply by all-solutions** [reply]
The SMH-II series should work. 
The list of confirmed devices only includes the "normal" SMH, but I think someone on Discord recently said that the SMH-II also works.

Was a CD with the "WatchPower" or "SolarPower" software included with the delivery? If so, then it is compatible.

**Reply by paolopa65** [reply]
Thanks for the replies, it didn't come with any software, but it works with WatchPower and Energy-Mate.
I made everything as per the diagram, but I found a difference on the connector in the diagram there are only 3 wires connected to the inverter port, while if I look at its original connector with the wifi module there are 4 wires connected. I'm trying but I can't get any readings.

**Reply by all-solutions** [reply]
The fourth wire is the power supply for the Wifi module. You should not play with this, it has 12-15V on it.

Take a few photos where we can see the exact wiring and post them here.

**Comment by paolopa65** [comment]
![20240503_181436](https://github.com/softwarecrash/Solar2MQTT/assets/79114433/b2b59a0a-0cea-4078-b13e-489d7ba908e7)

original wifi module cable
1 red
2 white
4 black
8 green

**Reply by all-solutions** [reply]
Not from the plug, from your cabling Inverter -> TTL/RS232 -> Wemos ;-)

BTW: Pin 4 is the 12-15V wire.

**Comment by paolopa65** [comment]
TTL/RS232 -> Wemos

VCC           -> 3v3 yellow
Gnd           -> g black
Rxd            -> D6 brown
Txd             -> White D7   
then from the TTL/RS232 connector to the RJ-45 as shown in the diagram.   
I also tried connecting its original cable to TTL/RS232 but without success.  
![1](https://github.com/softwarecrash/Solar2MQTT/assets/79114433/8ec35830-1019-43ea-a290-0fb24bf98f6e)
![2](https://github.com/softwarecrash/Solar2MQTT/assets/79114433/94f718f8-4b97-449b-97c4-70dbcee682bb)
![3](https://github.com/softwarecrash/Solar2MQTT/assets/79114433/aa5365ea-4b6d-4b61-9d35-2430d6d2f4dd)

**Reply by all-solutions** [reply]
And you are using the original cable supplied with the inverter?
If so, this will not work, as the RX and TX are then reversed.
This is why the circuit diagram on Github is also shown with direct (soldered) wires on the PCBs.

What you can do: You could use a null-modem-gender changer. This then turns RX and TX. 
In theory it should work, but I have never tested it myself.

![image](https://github.com/softwarecrash/Solar2MQTT/assets/17761850/6c57eef4-8a6f-4f2c-a577-04c043130245)
https://amzn.eu/d/cXvI0Wp

**Comment by paolopa65** [comment]
No, I tried the original cable, but it doesn't work as you said.
I redid the cable as per the diagram and I also tried to reverse the Rx and Tx dupond connectors on the RS232 but nothing continues to appear as readings
![4](https://github.com/softwarecrash/Solar2MQTT/assets/79114433/0fc20927-d877-478e-b580-a0a17bfa20a8)

**Comment by paolopa65** [comment]
Works!!!!!!!!
Damn TTL/RS232 module was broken!!
I changed the TTL/RS232 module with another one and now I see all the data!!
Thanks for all the help offered, kind regards.

**Reply by all-solutions** [reply]
Perhaps you have killed the module with the original cable. There are 12V on the DB9 connector.

But it's great that it's working now.

**Comment by paolopa65** [comment]
It would be possible to put a simple RJ45 splitter to be able to connect its original cable and the wifi module for Watchpower and the cable prepared for Solar2mqtt.
The type in the photo.
![5](https://github.com/softwarecrash/Solar2MQTT/assets/79114433/2b3207ed-e81b-4ed9-ae49-10aa2fbafc82)

**Reply by all-solutions** [reply]
No, that is not possible. RS232 is a point-to-point protocol, not a point-to-multipoint.
Imagine you (as Inverter) are talking to person A (Solar2MQTT) and person B (Wifi-Dongle) is chatting in between. Then you don't understand either of them.

BTW: those "splitters" you have in the photo are anything but not "real" splitters. 98% of them have this pinout:
![image](https://github.com/softwarecrash/Solar2MQTT/assets/17761850/83dc51b9-49db-4afa-abba-36853380b711)
...and are designed to supply 2 computers via a network cable. To do this, however, a "splitter" must be installed on the router.
![image](https://github.com/softwarecrash/Solar2MQTT/assets/17761850/2007cacd-129f-4ce3-8f88-48fabbba28ff)
